inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/source/AbstractSourceOperator.java

```diff
@@ -162,10 +162,11 @@ public void updateOpt(SourceRequest request, Integer groupStatus, String operato
         entity.setModifyTime(new Date());
 
         // Re-issue task if necessary
+        entity.setPreviousStatus(entity.getStatus());
         if (GroupStatus.forCode(groupStatus).equals(GroupStatus.CONFIG_SUCCESSFUL)) {
-            LOGGER.info("re-issuing source task {}, previous status = {}", entity.getId(), entity.getStatus());
-            entity.setPreviousStatus(entity.getStatus());
             entity.setStatus(SourceStatus.TO_BE_ISSUED_ADD.getCode());
+        } else {
+            entity.setStatus(SourceStatus.SOURCE_NEW.getCode());
```

Comment:
Not `CONFIG_SUCCESSFUL`, code will set `SourceStatus.SOURCE_NEW`.   Do we need  set `SourceStatus.SOURCE_NEW` by judging status?

Reply:
Do you have any ideas?

Reply:
Limit the original source status range to be either success or fail. Other status leave unchanged.